### PR TITLE
(789) - Apply - Submit application

### PR DESCRIPTION
**Pull request by richpjames**
# Context
We need to submit completed applications to the API.
The word submission is perhaps misleading, maybe 'finalize' would be more accurate as we are only 'submitting' the Application ID as the API already holds all the data it needs. The user simply tells it the application is complete.

As part of this work I have added a confirmation screen as it was a small piece of work and we needed somewhere to redirect to.

[Submission Trello card](https://trello.com/c/Gjsg2teU/789-persist-submitted-documents-in-translated-json)
[Confirmation screen Trello card](https://trello.com/c/I8ngVZL4/657-confirmation-screen)

# Changes in this PR
We add the required methods to the client, service and controller and wire up the 'Submit application' button to confirm completion of the application to the API

## Screenshots of UI changes
Apply -- allows completion of the form

### CODE REVIEW

**Review by pezholio — COMMENTED** [review]
Cool - nice work! I think it might be a good opportunity to confirm the data we expect is sent to the application - a similar pattern to https://github.com/ministryofjustice/hmpps-approved-premises-ui/blob/main/integration_tests/tests/manage/arrivals.cy.ts#L42

File: server/controllers/apply/applicationsController.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export default class ApplicationsController {
       if (crnArr.length) {
         const person = await this.personService.findByCrn(req.user.token, crnArr[0])
 
-        return res.render(`applications/confirm`, {
+        return res.render(`applications/personConfirmation`, {
```
Comment (pezholio) [inline server/controllers/apply/applicationsController.ts:52]: I think `applications/people/confirm` might be better naming here

File: server/controllers/apply/applicationsController.ts [hunk 2]
```diff
@@ -78,4 +79,16 @@ export default class ApplicationsController {
       )
     }
   }
+
+  submit(): RequestHandler {
+    return async (req: Request, res: Response) => {
+      const application = await this.applicationService.findApplication(req.user.token, req.params.id)
+      application.document = getResponses(application)
+
+      await this.applicationService.saveToApi(application, req)
```
Comment (pezholio) [inline server/controllers/apply/applicationsController.ts:88]: Do we need to save to the API and submit here?

File: server/views/applications/submissionConfirmation.njk [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    <div class="govuk-grid-row">
+        <div class="govuk-grid-column-two-thirds">
+            {{ govukPanel({
+                titleText: "Application confirmation"
+            }) }}
+            <p>
+          We’ve sent you a confirmation email.
+        </p>
+            <p>
+                <a href="">Download a copy of this application (PDF, 708 KB) </a>
```
Comment (pezholio) [inline server/views/applications/submissionConfirmation.njk:21]: I missed this from the prototypes. Will be very cool to be able to do this. Should be fairly easy too!

File: server/views/applications/submissionConfirmation.njk [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            </p>
+
+            <h2 class="govuk-heading-m">What happens next</h2>
+
+            <p>We’ve sent your application to the Midlands central referral unit (CRU) for Approved Premises (AP).</p>
```
Comment (pezholio) [inline server/views/applications/submissionConfirmation.njk:26]: Not for now, but I think we'll need to get the CRU from the API response

**Review by pezholio — APPROVED** [review]
LGTM - just some minor bits

File: server/controllers/apply/applicationsController.ts (hunk 2 above)
Comment (richpjames) [reply server/controllers/apply/applicationsController.ts:88]: We probably don't have to... 
But we only submit the translated document.
By saving it we save the application with the translated text so the API then has the final version of the full Application object. 
But the API must be doing some work to attatch the translated text to the application object anyway

**Pull request merged**